generate-version-and-release-notes.js

```diff
@@ -135,26 +137,44 @@ function getUpdatedVersion(version, changes) {
   }
 }
 function getChange(line, commitTypes) {
-  let type = "";
-  let content = "";
-
-  if (line.split(":").length >= 2) {
-    const prefix = line.split(":")[0].trim();
-    content = line.split(":")[1].trim();
-    const validPrefix = commitTypes.some(x => x.commitType == prefix);
-
-    type = validPrefix ? commitTypes
-      .find(x => x.commitType == prefix)
-      .releaseType : "none"
-
-  } else {
-    type = "none"
-    content = line.trim()
+  const convRegex = /(?<type>^[a-z]+)(?<scope>\([a-z\d,\-]+\))?(?<breaking>!)?(?<colon>:{1})(?<space> {1})(?<subject>.*)/;
+  let matchResult = line.match(convRegex);
+
+  if (matchResult) {
+
+    let result = {
+      type: '',
+      content: ''
+    }
+    let { type, breaking, scope, subject } = matchResult.groups;
+    breaking = breaking == '!'
+    const validPrefix = commitTypes.some(x => x.commitType == type);
+
+    if (validPrefix) {
+      result.type = breaking ? "major" : commitTypes.find(x => x.commitType == type).releaseType
+      result.content = subject.trim()
+      if (scope)
+        result.content = `${scope.trim()}: ${result.content}`
+    } else {
+      result.type = "none"
+      result.content = line.trim()
+    }
```

Comment:
changelog example:
https://github.com/ohpen/platform-repository-for-testing/blob/main/CHANGELOG.md

![image](https://user-images.githubusercontent.com/97619732/162962558-45825469-050c-46c8-b128-b02cd06be36d.png)